### DISCUSSION TITLE
Is there a way to convert timestamp to local time on output?

### DISCUSSION THREAD

**Discussion by 41405465** [vectordotdev/vector#18368] [Q&A]
The event timestamp can only be the UTC timestamp, I want to be able to output the local timestamp, for example, through the "internal_logs" data source, the vector log data is output to a file, the timestamp of each log in the file is still UTC time, is there a way to convert the output local time, match the time of the machine.

As follows:
{"host":"DEV-DFNCFT-AP08","message":"Signal received.","metadata":{"kind":"event","level":"INFO","module_path":"vector::signal","target":"vector::signal"},"pid":46992,"signal":"SIGINT","source_type":"internal_logs","timestamp":"**2023-08-25T07:09:04.787290941Z**"}

**Comment by jszwedko** [comment]
I think you could use https://vector.dev/docs/reference/vrl/functions/#format_timestamp to convert the timestamp to a string in a given timezone, before the `file` sink.

https://github.com/vectordotdev/vector/issues/18158 is tracking the ability to configure timezones at the sink level.
